### PR TITLE
Apply contract read value hook to v1 and v2 contract readers [3/n]

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

Apply contract read value hook to v1 and v2 contract reader hooks.

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1869** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1869" target="_blank">Graphite</a> 
    * **PR #1870** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1870" target="_blank">Graphite</a> 
      * **PR #1871** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1871" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1871?utm_source=stack-comment).

File: src/hooks/v1/contractReader/ContractReader.ts [hunk 1]
```diff
@@ -31,72 +31,30 @@ export default function useContractReader<V>({
...
 
-    getValue()
+  // Call the callback on contract read value changed
+  useDeepCompareEffectNoCheck(() => {
+    _callback(value)
+  }, [value])
 
-    const listener = () => getValue()
+  useDeepCompareEffectNoCheck(() => {
+    const listener = () => refetchValue()
```
Comment (wraeth-eth) [inline src/hooks/v1/contractReader/ContractReader.ts:57]: It is also possible we can pull this stuff out into a separate hook, but I will leave that for another day

File: src/hooks/v1/contractReader/ContractReader.ts [hunk 2]
```diff
@@ -31,72 +31,30 @@ export default function useContractReader<V>({
...
-        setValue(_formatter(undefined))
-        _callback(_formatter(undefined))
-      }
-    }
+  const { value, refetchValue } = useContractReadValue({
+    contract,
+    contracts,
+    functionName,
+    args,
+    version: 'V1',
```
Comment (peripheralist) [inline src/hooks/v1/contractReader/ContractReader.ts:46]: is the only thing differentiating v1 and v1 contractReaders this `version` arg?

if so i'd be down to remove it. not sure the logging is worth the added complexity, and like i mentioned in an another comment there should also be some reads where version isn't defined

Comment (wraeth-eth) [reply src/hooks/v1/contractReader/ContractReader.ts:46]: ah totally re where version shouldn't be defined.

I think I tend to agree that the version here isn't adding much to the logging so we can remove it.

**Pull request closed without merging**